### ISSUE TITLE
Take into account “default-server” statement for servers created with runtime API.

### ISSUE THREAD

**Issue by cpaillet** [haproxy/haproxy#1994]
### Your Feature Request

When we create a server using the runtime api, the `default-server` statements (from the global configuration or from backend configuration) is not taken into account.

According to the documentation [add server configuration](https://docs.haproxy.org/2.7/management.html#add%20server) "_Also note that no settings will be reused from an hypothetical 'default-server' statement in the same backend._" it seems expected, but many other sections aren't clear about the fact that `default-server` is completely ignored when using the server addition API. 
The docs don’t say if it’s working or not for global `default-server`, but my observation seems it’s not.

My request is to have all the keywords supported by the Runtime API and that those set in `default-server` sections are taken in account when we add a server, if not overridden.

### What are you trying to do?

If the configuration is
```
defaults
    default-server slowstart 20s weight 20
backend back1
     default-server weight 200
     server srv1 10.0.0.1:80
```
and if I add a server using
 `echo "add server back1/srv2 10.0.0.2:80" | socat stdio  /tmp/haproxy.sock`
I want both servers have the same configuration, i.e slowstart 20s & weight 200

### Output of `haproxy -vv`

```plain
tested in haproxy 2.7
```

**Comment by a-denoyelle** [comment]
Indeed, currently the documentation should be completed `from an hypothetical 'default-server' statement in the same backend` with `or from a default or global section`.

Our reflexion is that dynamic server API should be mainly used with automation tool. In this context, it should be not too complicated to always use explicit argument. The main reason is to avoid unexpected settings coming from `default-server`. While it should be easy enough to review `default-server` while writing the configuration by hand, it may be easy to forget such a statement when tweaking new servers via an external tool through the runtime API.

**Comment by cpaillet** [comment]
In our case, the automation tools for dynamic servers, are not totally aware of all the params. His "job" it's just to get a list of servers from the inventory (consul) for a given backend and configure it. it don't know all the global parameters or backend parameters.

for us, it's also very disturbing to not have the same behavior when writing the configuration in the file or with api. ( We spend a few days to understand why servers created by the configuration file and those added later by the API didn't have the same behavior ).

A configuration with global parameters, backend parameters, it's easier to review for a human, we can check easily what is the global parameters, backend parameters, and don't forget something.

and the last point, it's not possible to get all the server params with `show servers state xxx`. We want to keep the running configuration auditable at every time. Then we don't want to change any param with the API we cannot retrieve. ( and then, keep those params at backend/global level )
